lib/rules/require-expect.js

```diff
@@ -0,0 +1,113 @@
+        return callee.object &&
+               callee.object.type === "Identifier" &&
+               callee.object.name === currentTest.assertName &&
+               callee.property.name === "expect";
+    }
+
+    function isTopLevelExpectCall(callee) {
+        return isExpectCall(callee) && currentTest.blockDepth === 1;
+    }
+
+    function isUsingAssertInNestedBlock(node) {
+        return currentTest.blockDepth > 1 && utils.isAssertion(node.callee, currentTest.assertName);
+    }
+
+    function isPassingAssertAsArgument(node) {
+        for (var i = 0; i < node.arguments.length; i++) {
+            if (node.arguments[i].name === currentTest.assertName) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    function isViolatingExceptSimpleRule(node) {
+        return (isUsingAssertInNestedBlock(node) || isPassingAssertAsArgument(node)) &&
+                !currentTest.isExpectUsed;
+    }
+
+    function captureTestContext(node) {
+        currentTest = {
+            assertName: utils.getAssertContextNameForTest(node.arguments),
+            blockDepth: 0,
+            isExpectUsed: false
+        };
+    }
+
+    function releaseTestContext() {
+        currentTest = false;
+    }
+
+    var ExceptSimpleStrategy = {
+        "CallExpression": function (node) {
+            if (currentTest) {
+                if (isTopLevelExpectCall(node.callee)) {
+                    currentTest.isExpectUsed = true;
+                } else if (isViolatingExceptSimpleRule(node)) {
+                    context.report({ node: node, message: EXCEPT_SIMPLE_ERROR_MESSAGE });
+                }
+            } else if (utils.isTest(node.callee)) {
+                captureTestContext(node);
+            }
+        },
+
+        "CallExpression:exit": function (node) {
+            if (utils.isTest(node.callee)) {
+                releaseTestContext();
+            }
+        },
+
+        "BlockStatement": function () {
+            if (currentTest) {
+                currentTest.blockDepth++;
+            }
+        },
+
+        "BlockStatement:exit": function () {
+            if (currentTest) {
+                currentTest.blockDepth--;
+            }
+        }
+    };
+
+    var AlwaysStrategy = {
+        "CallExpression": function (node) {
+            if (currentTest && isExpectCall(node.callee)) {
+                currentTest.isExpectUsed = true;
+            } else if (utils.isTest(node.callee)) {
+                captureTestContext(node);
+            }
+        },
+
+        "CallExpression:exit": function (node) {
+            if (utils.isTest(node.callee)) {
+                if (!currentTest.isExpectUsed) {
+                    context.report({ node: node, message: ALWAYS_ERROR_MESSAGE });
+                }
+
+                releaseTestContext();
+            }
+        }
+    };
+
+    return context.options[0] === "except-simple" ? ExceptSimpleStrategy : AlwaysStrategy;
+};
```

Comment:
Could you add a schema? I think you could do something like this:

```js
// End of rule here...

module.exports.schema = {
    type: "array",
    items: [
        enum: ["always", "except-simple"]
    ],
    minItems: 0,
    maxItems: 1
};
```

This allows ESLint to validate configuration at runtime, before running the rule. In other words, it will fail faster if users have bad configuration.